ui/src/reducers/session.js

```diff
@@ -29,4 +31,9 @@ const handleLogout = state => ({
 export default createReducer({
   [loginWithToken]: handleLogin,
   [logout]: handleLogout,
+  [fetchMetadata.COMPLETE]: (state, {metadata}) => {
+    if (!metadata?.token) {return state}
```

Comment:
Nice. You could also build a check into handleLogin to  check if it gets a token, keep the code in there rather than here.

Reply:
done